src/devices/Arduino/ArduinoGpioControllerDriver.cs

```diff
@@ -14,26 +14,28 @@ namespace Iot.Device.Arduino
 {
     internal class ArduinoGpioControllerDriver : GpioDriver
     {
-        private readonly ArduinoBoard _arduinoBoard;
+        private readonly FirmataDevice _device;
         private readonly IReadOnlyCollection<SupportedPinConfiguration> _supportedPinConfigurations;
         private readonly Dictionary<int, CallbackContainer> _callbackContainers;
         private readonly ConcurrentDictionary<int, PinMode> _pinModes;
         private readonly object _callbackContainersLock;
         private readonly AutoResetEvent _waitForEventResetEvent;
         private readonly ILogger _logger;
+        private readonly ConcurrentDictionary<int, PinValue?> _outputPinValues;
```

Comment:
I'm not following why do you need this ConcurrentDictionary. not sure why you want to track the values, but this will have a significant perf hit because ConcurrentDictionary uses locks which of course slow perf overall.

Reply:
This is used to check whether a given pin is already set to a certain value. Accessing this dictionary should be significantly faster than what it uses to send a command.